packages/next/server/app-render.tsx

```diff
@@ -1572,7 +1583,7 @@ export async function renderToHTMLOrFlight(
   )
 
   const requestStore = {
-    headers: new ReadonlyHeaders(headersWithoutFlight(req.headers)),
+    headers: lazy(() => new ReadonlyHeaders(headersWithoutFlight(req.headers))),
```

Comment:
Instead of creating a `lazy` method, IMHO it is possible to simply define a getter on the `requestStore` object to achieve lazy evaluation.


```suggestion
    get headers() { return new ReadonlyHeaders(headersWithoutFlight(req.headers))) }
```

Reply:
I initially went for that at first but this means that we would create a new instance of the headers every time I think we call `headers()` so I went with the lazy/singleton approach in order to not mess with that

Reply:
```js
let cachedHeaderInstance: ReadonlyHeaders | undefined;
const requestStore = {
  get headers() {
    cachedHeaderInstance ||= new ReadonlyHeaders(headersWithoutFlight(req.headers)));
	  return cachedHeaderInstance
	},
```

This could achieve both lazy and singleton, without changing the API interface.

Reply:
you're right it slipped my mind 🫣

Reply:
Also, any reason not to apply the same approach on the `cookies` as well? I thought most users won't access `cookies` *on every request*.